tests/test/org/commcare/backend/util/test/SessionNavigatorTests.java

```diff
@@ -0,0 +1,166 @@
+package org.commcare.backend.util.test;
+
+import org.commcare.suite.model.SessionDatum;
+import org.commcare.test.utilities.MockSessionNavigationResponder;
+import org.commcare.test.utilities.MockApp;
+import org.commcare.util.SessionNavigator;
+import org.commcare.util.mocks.SessionWrapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests for SessionNavigator.java
+ *
+ * @author amstone
+ */
+public class SessionNavigatorTests {
+
+    MockApp mApp;
+    MockSessionNavigationResponder mSessionNavigationResponder;
+    SessionNavigator sessionNavigator;
+
+    @Before
+    public void setUp() throws Exception {
+        mApp = new MockApp("/session-tests-template/");
+        mSessionNavigationResponder = new MockSessionNavigationResponder(mApp.getSession());
+        sessionNavigator = new SessionNavigator(mSessionNavigationResponder);
+    }
+
+    @Test
+    public void testNavWithoutAutoSelect() {
+        SessionWrapper session = mApp.getSession();
+
+        // Before anything has been done in the session, the sessionNavigatorResponder should be
+        // directed to get a command
+        sessionNavigator.startNextSessionStep();
+        Assert.assertEquals(SessionNavigator.GET_COMMAND,
+                mSessionNavigationResponder.getLastResultCode());
```

Comment:
This code block is repeated several times, would be good to lift into a helper or put once in the `setUp` method